src/xml_parsing.cpp

```diff
@@ -46,6 +46,188 @@ auto StrEqual = [](const char* str1, const char* str2) -> bool {
   return strcmp(str1, str2) == 0;
 };
 
+namespace
+{
+/**
+ * @brief Throws a BT::RuntimeError with an error message a line number and some text
+ */
+void ThrowError(int line_num, const std::string& text)
+{
+  char buffer[256];
+  sprintf(buffer, "Error at line %d: -> %s", line_num, text.c_str());
+  throw BT::RuntimeError(buffer);
+}
+
+/**
+ * @brief Returns the number of children under an XML node
+ */
+int ChildrenCount(const XMLElement* parent_node)
+{
+  int count = 0;
+  for (auto node = parent_node->FirstChildElement(); node != nullptr;
+       node = node->NextSiblingElement())
+  {
+    count++;
+  }
+  return count;
+}
+
+/**
+ * @brief Recursively enforces that the attributes of an XML node and all its child nodes are all correct.
+ * @throw BT::RuntimeError if an error was found.
+ */
+void verifyXMLRecursively(
```

Comment:
DO not move this code (hard for me to compare) and keep name and namespace

Reply:
I reverted my changes to these functions to keep the inline helper functions, which reduces the size of the diff.